config/opal_setup_cc.m4

```diff
@@ -23,6 +23,119 @@ dnl
+                   [[#define FOO(x) (_Generic (x, int: 1))]], [[static int x, y; y = FOO(x);]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Static_assert], [opal_prog_cc_c11_helper__static_assert_available],
+                   [[#include <stdint.h>]],[[_Static_assert(sizeof(int64_t) == 8, "WTH");]])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE_C11_ATOMIC], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports C11 atomics])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__atomic_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__ATOMIC], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports __Atomic])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__thread_local_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__THREAD_LOCAL], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports __Thread_local])
+
+    dnl At this time Open MPI only requires thread local and the atomic convenience types
+    AS_IF([test "x$opal_prog_cc_c11_helper__thread_local_available" = "xyes" && test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"],
+          [$2], [$3])
+
+    CFLAGS=$opal_prog_cc_c11_helper_CFLAGS_save
+
+    OPAL_VAR_SCOPE_POP
+])
+
+AC_DEFUN([OPAL_PROG_CC_C11],[
+    OPAL_VAR_SCOPE_PUSH([opal_prog_cc_c11_flags])
+    if test -z "$opal_cv_c11_supported" ; then
+        opal_cv_c11_supported=no
+        opal_cv_c11_flag_required=yes
+
+        AC_MSG_CHECKING([if $CC requires a flag for C11])
+
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#if __STDC_VERSION__ < 201112L
+#error "Need a flag for C11"
+#endif
+                                           ]],[])],
+                          [opal_cv_c11_flag_required=no])
+
+        AC_MSG_RESULT([$opal_cv_c11_flag_required])
+
+        if test "x$opal_cv_c11_flag_required" = "xno" ; then
+            AC_MSG_NOTICE([verifying $CC supports C11 without a flag])
+            OPAL_PROG_CC_C11_HELPER([], [], [opal_cv_c11_flag_required=yes])
+        fi
+
+        if test "x$opal_cv_c11_flag_required" = "xyes" ; then
+            opal_prog_cc_c11_flags="-std=gnu11 -std=c11 -c11"
+
+            AC_MSG_NOTICE([checking if $CC supports C11 with a flag])
+            opal_cv_c11_flag=
+            for flag in $(echo $opal_prog_cc_c11_flags | tr ' ' '\n') ; do
+                OPAL_PROG_CC_C11_HELPER([$flag],[opal_cv_c11_flag=$flag],[])
+                if test "x$opal_cv_c11_flag" != "x" ; then
+                    CFLAGS="$CFLAGS $opal_cv_c11_flag"
+                    AC_MSG_NOTICE([using $flag to enable C11 support])
+                    opal_cv_c11_supported=yes
+                    break
+                fi
+            done
```

Comment:
Might want to put a comment here that it's (at least for now) not an error if C11 is not supported.

Reply:
Added.